transport/src/commonMain/kotlin/com/genesys/cloud/messenger/transport/core/MessagingClient.kt

```diff
@@ -83,12 +83,24 @@ interface MessagingClient {
      *
      * @throws IllegalStateException
      */
+    @Deprecated("Use the connect(shouldConfigure: Boolean) instead", ReplaceWith("connect(shouldConfigure: Boolean)"))
     @Throws(IllegalStateException::class)
     fun connect()
 
+    /**
+     * Open a secure WebSocket connection to the Web Messaging service with the url and
+     * deploymentId configured on this MessagingClient instance.
+     *
+     * @param shouldConfigure when true, Configure a Web Messaging session.
+     * @throws IllegalStateException
+     */
+    @Throws(IllegalStateException::class)
+    fun connect(shouldConfigure: Boolean = false)
```

Comment:
Just wondering if `shouldConfigure` should default to `true` instead of `false`.

Also wondering about the parameterization of configuring the session in this connect method. Is the idea to be able to preserve the current functionality in the future, when `connect()` and `configureSession()` are removed. If so and the SDK user wanted the same behavior today being able to separate socket connection and session configuration, would they use this interface as below?

```
// if trying to keep behavior in future same as today, does this:
connect()
// ...
configureSession()

// become this:

connect(shouldConfigure = false)
// ...
connect(shouldConfigure = true)
```